### ISSUE TITLE
Staging Review finding: Error alert plus duplicative content plus focus management

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#98479]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** IIR
**Product:** Veteran New Account Flow within Authenticated Experience
**Feature:** Welcome Pathway V1

## Findings details
**VA.gov Experience Standard - issue:** [User encounters unnecessary repetition or redundancy.](https://design.va.gov/about/experience-standards/#user-encounters-no-repetition-or-redundancy)
**VA.gov Experience Standard - category:** [Efficiency](https://design.va.gov/about/experience-standards/#category-efficiency)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

On [/review-information/contact-information](https://staging.va.gov/my-va/welcome-va-setup/review-information/contact-information) when a new user doesn't have a mobile phone number saved but presses Continue anyway, two things happen:
- An error alert is added above the warning alert that was already there.
- Focus is sent to a `div` that contains the "Note" paragraph, the error alert, and the warning alert.

The error alert includes `role="alert"` out of the box, which means screen readers will announce the content of that alert before anything else as soon as it's added to the page. Moving focus to the `div` means the contents of that `div` will be announced immediately after the new `role="alert"` content. The net result is that screen readers will announce the error alert, the "Note" paragraph, the error alert again, and the warning alert --- all in one immediate flow.

In other words, if you're a screen reader user with no mobile phone provided and you click Continue, you'll hear:

> We still don't have your mobile phone. Please edit and update the field. Note: A mobile phone is required for this application. We still don't have your mobile phone. Please edit and update the field. Your mobile phone is missing. Please edit and update the field.

The combination of the error alert plus duplicative content plus focus management results in a lot of repetition. If you're a screen reader user and you don't have the visual cues for the different styling of content on the page and you hear four repeated and nearly-identical reminders about your mobile phone, it's easy to conclude that something broken or you're stuck in some kind of loop. This is especially true for a blind Veteran with a recently-acquired disability who's new to using screen readers and who's trying to access VA benefits online for the first time. When you're learning assistive technology (and learning to use a computer in a wholly new way), there's a pretty common impulse to assume it's just not working, and not assume that what you're hearing is what's intended.

<details><summary>Link, screenshot or steps to recreate</summary>

![Add your contact information page, with focus indicator around "Note" paragraph, "We still don't have" error alert, and "Your mobile phone number is missing" warning alert.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/82257-Brian-2.png?raw=true)

</details>

## Recommended action

I don't think it makes sense to have both the warning alert and the error alert, and I don't think the focus management is necessary here. If you are moving focus, it should be more tightly managed, rather than to a wrapper containing duplicative content.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by kingbmike** [comment]
@briandeconinck Our team had a meeting with the Forms team and @humancompanion-usds (Matthew Dingee) where we decided that the error stacking that is currently in prod needs remedied, but is Platform code.  Before 'fixing' this issue, I had a meeting with @Mottie (Robin Garrison) and @jeana-adhoc (Jeana Clark) knowing that it would have downstream effects on at least Robin's form as well.  We came to the conclusion that the Forms team should design the build for the remedy to this issue and Robin offered to implement once completed.

Being that this Platform code is already in prod, I'm hoping that you'll consider this not a launch blocker any longer and allow us to deploy before this is addressed by the Forms team?

Also, ['Role="Alert" Wrapping Warning Alert'](https://github.com/department-of-veterans-affairs/va.gov-team/issues/98478) is the other launch blocking issue we have that is Platform Code as well.  Robin's form also has this therefore more downstream effects, etc.  I would ask that this issue also be considered to be addressed by the forms team and since already in prod, not a launch blocking issue for our form right now.

Thank you for your time and consideration with both of these!

**Comment by briandeconinck** [comment]
Hey @kingbmike! Thank you for the update, and thank you for the tremendous effort on your end to try to work through these issues!

As far as I'm concerned, this all sounds good --- efforts are underway to resolve, and I think in this case you're okay to go ahead and launch. I'm removing the `launch-blocking` label but keeping the `a11y-defect-1` label on both of these tickets for tracking purposes. cc @artsymartha68

**Comment by kingbmike** [comment]
Sounds like a plan - Thank you, @briandeconinck

**Comment by kingbmike** [comment]
This is a form wide - platform code issue.
